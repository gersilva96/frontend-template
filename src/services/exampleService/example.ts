import urlJoin from "url-join";
import { CONFIG } from "~/config";
import { request } from "~/services/common";
import { HTTPMethods, RequestResponse } from "~/types/request";
import { GenericObject } from "~/types/utils";

const exampleApiBaseUrl = CONFIG.SERVICES.EXAMPLE.BASE_URL as string;
const exampleBasePath = "/example";

export const getSomethingById = (id: string): RequestResponse<string> => {
  const url = urlJoin(exampleApiBaseUrl, exampleBasePath, id);
  return request({
    method: HTTPMethods.GET,
    url
  });
};

export const postSomething = (
  something: GenericObject
): RequestResponse<string> => {
  const url = urlJoin(exampleApiBaseUrl, exampleBasePath);
  return request({
    method: HTTPMethods.POST,
    data: something,
    url
  });
};
