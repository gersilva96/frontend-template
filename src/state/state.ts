export enum SliceNames {
  Example = "Example"
}
