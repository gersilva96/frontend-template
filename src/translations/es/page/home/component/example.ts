export const example = {
  input: {
    label: "Ingresar algo",
    placeholder: "Ej. programaci√≥n"
  },
  submitButton: "Setear",
  value: 'El valor de "something" es: {{value}}'
};
