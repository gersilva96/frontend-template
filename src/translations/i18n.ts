import i18next from "i18next";
import { initReactI18next } from "react-i18next";
import { es } from "./es/es";

const resources = {
  es: {
    translation: {
      ...es
    }
  }
};

i18next.use(initReactI18next).init({
  resources,
  lng: "es",
  initImmediate: true,
  interpolation: {
    escapeValue: false
  }
});

export const { t } = i18next;

const tkES = i18next.getDataByLanguage("es")
  ?.translation as unknown as typeof es;

export const tkCommon = tkES.common;
export const tkHome = tkES.page.home;

export default i18next;
