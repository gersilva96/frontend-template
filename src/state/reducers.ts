import { exampleSlice } from "./features/example/slice";
import { SliceNames } from "./state";

export const reducers = {
  [SliceNames.Example]: exampleSlice.reducer
};
