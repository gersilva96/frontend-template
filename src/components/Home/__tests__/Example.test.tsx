import { act, fireEvent, render, screen } from "@testing-library/react";
import i18next from "i18next";
import {
  AppProviders,
  AppProvidersProps
} from "~/components/common/AppProviders/AppProviders";
import { somethingSelector } from "~/state/features/example/selectors";
import { store } from "~/state/store";
import { t, tkHome } from "~/translations/i18n";
import { Example } from "../Example/Example";

describe("Example component", () => {
  let tk: typeof tkHome.component.example;
  let providersProps: AppProvidersProps;

  beforeAll(() => {
    tk = tkHome.component.example;
    providersProps = { store, i18n: i18next };
  });

  it("should match with the snapshot", () => {
    const { container } = render(
      <AppProviders {...providersProps}>
        <Example />
      </AppProviders>
    );
    expect(container).toMatchSnapshot();
  });

  it("should have the input value empty at the beginning", () => {
    render(
      <AppProviders {...providersProps}>
        <Example />
      </AppProviders>
    );
    const inputPlaceholderText = t(tk.input.placeholder);
    const input: HTMLInputElement =
      screen.getByPlaceholderText(inputPlaceholderText);
    expect(input.value).toBe("");
  });

  it('should have the input value "algo" after change it', async () => {
    render(
      <AppProviders {...providersProps}>
        <Example />
      </AppProviders>
    );
    const inputPlaceholderText = t(tk.input.placeholder);
    const input: HTMLInputElement =
      screen.getByPlaceholderText(inputPlaceholderText);
    await act(async () => {
      fireEvent.input(input, { target: { value: "algo" } });
    });
    expect(input.value).toBe("algo");
  });

  it('should have the redux state value "algo" after change the input value and clicking submit button', async () => {
    render(
      <AppProviders {...providersProps}>
        <Example />
      </AppProviders>
    );
    const something = somethingSelector(providersProps.store.getState());
    expect(something).toBe("");
  });

  it("should have the redux state value empty at the beginning", async () => {
    render(
      <AppProviders {...providersProps}>
        <Example />
      </AppProviders>
    );
    const inputPlaceholderText = t(tk.input.placeholder);
    const input: HTMLInputElement =
      screen.getByPlaceholderText(inputPlaceholderText);
    const submitButtonText = t(tk.submitButton);
    const submitButton: HTMLButtonElement = screen.getByRole("button", {
      name: submitButtonText
    });
    await act(async () => {
      fireEvent.input(input, { target: { value: "algo" } });
      fireEvent.click(submitButton);
    });
    const something = somethingSelector(providersProps.store.getState());
    expect(something).toBe("algo");
  });
});
