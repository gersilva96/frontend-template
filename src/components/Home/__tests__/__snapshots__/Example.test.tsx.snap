// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example component should match with the snapshot 1`] = `
<div>
  <div
    class=""
  >
    <label>
      Ingresar algo
    </label>
    <input
      id="something"
      placeholder="Ej. programaciÃ³n"
      value=""
    />
    <button>
      Setear
    </button>
    <p>
      El valor de "something" es: 
    </p>
  </div>
</div>
`;
